gapic/src/main/com/google/gapid/models/Devices.java

```diff
@@ -278,6 +322,7 @@ public default void onCaptureDevicesLoaded() { /* empty */ }
     public final Device.Instance device;
     public final Service.DeviceTraceConfiguration config;
     public final TraceTargets targets;
+    public boolean validationStatus = false;
```

Comment:
This makes this class mutatable, which is not good, especially public mutatable like this.

I propose, instead, let's create a `DeviceValidation` class, which currently just contains a `boolean passed`. Then, add a method to the device model that does the validation. Something like:
`public ListenableFuture<DeviceValidation> getDeviceValidation(Path.Device device)`.
If the the skip flag is set, always return `DeviceValidation.PASSED`. We can then also add the cache handling in this method.

Also, if you use a `SingleInFlight` `RpcController` than it will make sure only one validation request is in flight.

Reply:
Created a new DeviceValidationInfo class and restructured the way validation requests are handled. The rpcController was SingleInFlight already and it is the same in the updated class as well. Could you please take a look at the new class now?